homeassistant/util/pressure.py

```diff
@@ -11,18 +11,20 @@
     PRESSURE_PA,
     PRESSURE_PSI,
     UNIT_NOT_RECOGNIZED_TEMPLATE,
+    PRESSURE_MMHG,
 )
 
 _LOGGER = logging.getLogger(__name__)
 
-VALID_UNITS = [PRESSURE_PA, PRESSURE_HPA, PRESSURE_MBAR, PRESSURE_INHG, PRESSURE_PSI]
+VALID_UNITS = [PRESSURE_PA, PRESSURE_HPA, PRESSURE_MBAR, PRESSURE_INHG, PRESSURE_PSI, PRESSURE_MMHG]
```

Comment:
Is anything using this code?

Reply:
No, there is no code yet that uses this.
But in the forum I met a lot of requests to add these units. And I myself am more familiar with them. That's why I added them.

Reply:
mmHg is usual pressure unit for CIS sitizens

Reply:
So if nothing is using it, why should it be added?

Reply:
The reason is that this (and only this) unit of pressure measurement is familiar to the CIS population (and this is more than 280 million people).
Yes, this unit is not currently used in the Home Assistant code. But in the forum, people regularly ask to add it, for example, to the weather frontend.

* https://community.home-assistant.io/t/pressure-units-in-the-weather-component-can-be-more-diverse/60944
* https://community.home-assistant.io/t/value-template/34950/6
* https://community.home-assistant.io/t/esphome-pressure-hpa-to-mmhg/171527
* etc.

What is the point of making “crutches” in separate modules when you can centrally add this unit to the core and solve the problem?

Reply:
Following your logic, then you need to throw PRESSURE_PSI out of the code - this unit is also not used anywhere in it.

Reply:
Just because something has done in the past, gives no reason to do it again.

- <https://community.home-assistant.io/t/esphome-pressure-hpa-to-mmhg/171527> -> Is about ESPHome, not Home Assistant.
- <https://community.home-assistant.io/t/value-template/34950/6> -> is about template calculations, but is not related to this PR.
- <https://community.home-assistant.io/t/pressure-units-in-the-weather-component-can-be-more-diverse/60944> -> The person adding an integration that needs this, could have provided the code you are now providing.

Reply:
Just because you live in a country where other standards are generally accepted, gives no reason to ignore the standards generally accepted in countries on the other sixth of all the Earth’s land.

Yes, that integration developer could theoretically suggest this code. But for some reason he did not. Maybe because I didn’t even think that it was possible? Or maybe because the conversion of pressure units was added to the core code later than he did his component?

Adding this unit will allow this and other developers to make their work easier in the future. It will allow in the future users of Home Assistant from the CIS countries to use this system with great comfort.

I am sure that a little time will pass and this code will be claimed by developers and will cease to be dead.

Reply:
By the way, I also found an example when a person needs reverse conversion:https://community.home-assistant.io/t/change-the-unit-in-bloomsky-sensors/35837

Reply:
We won't accept this PR until there's an active use case in the code.

Reply:
Here is a use-case out-of-code. I, as many others, want to customize output of any weather sensor, which I'm not using ATM partially because there is no pressure unit, used in my country. I could use a template, but it would be a lot nicer to put one small setting in one sensor to make it work like all other system.
Please, put yourself in our shoes and image if your country's proffered pressure/temperature/whatever unit was not supported. Let's imagine for a second, that HA was originally developed not in the US, but somewhere in Europe with metric system. I really don't think that a US citizen coming with such a small proposal adding their unit of temperature measurement, would be rejected so harshly.  There was a time, when there were **no** temperature units, and it **was** corrected.
As for code usage - there are CIS weather services, such as Gismeteo and Yandex Weather, providing pressure in mmHg, not in hPa. And integrations for them **could** have used mmHg for their operation. Please see [this integration's example code](https://github.com/Limych/ha-gismeteo/blob/master/custom_components/gismeteo/__init__.py#L246). This code **could** have used the proposed unit.

Reply:
This is about adding "dead" code to our codebase. As soon as there is a use case for it in our codebase, we are happy to accept and add it.

The integration you have linked it not the Home Assistant codebase. Feel free to open up a PR to add an integration to our codebase that add this and the new integration.